tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -701,7 +704,7 @@ public static function dataCallConnectOnce(): iterable
     public function testCallConnectOnce(string $method, array $params): void
     {
         $driverMock   = $this->createMock(Driver::class);
-        $pdoMock      = $this->createMock(Connection::class);
+        $pdoMock      = $this->createMock(PDO::class);
         $platformMock = $this->createMock(AbstractPlatform::class);
         $stmtMock     = $this->createMock(Statement::class);
 
```

Comment:
Not sure that your tests modification actually test your modification. 

I would rather add a functional test where the connection is not builded by DriverManager : https://github.com/doctrine/dbal/blob/1372a42083dedc56016051d61526507fb5521f9c/tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php#L354-L363

Reply:
Added